File: kmp/composeApp/src/commonMain/kotlin/com/bloom/core/utils/DatePrecision.kt
package com.bloom.core.utils

import kotlinx.datetime.LocalDateTime

enum class DatePrecision { EXACT, MONTH, QUARTER }

private val FRENCH_MONTHS = arrayOf(
    "janvier", "février", "mars", "avril", "mai", "juin",
    "juillet", "août", "septembre", "octobre", "novembre", "décembre"
)

fun formatDateFr(date: LocalDateTime, precision: DatePrecision): String = when (precision) {
    DatePrecision.EXACT -> "${date.dayOfMonth} ${FRENCH_MONTHS[date.monthNumber - 1]} ${date.year}"
    DatePrecision.MONTH -> "${FRENCH_MONTHS[date.monthNumber - 1]} ${date.year}"
    DatePrecision.QUARTER -> {
        val q = (date.monthNumber - 1) / 3 + 1
        "T$q ${date.year}"
    }
}

fun datePrecisionFromString(s: String?): DatePrecision = when (s) {
    "month" -> DatePrecision.MONTH
    "quarter" -> DatePrecision.QUARTER
    else -> DatePrecision.EXACT
}

fun datePrecisionToString(p: DatePrecision): String = when (p) {
    DatePrecision.EXACT -> "exact"
    DatePrecision.MONTH -> "month"
    DatePrecision.QUARTER -> "quarter"
}

File: lib/core/utils/date_precision.dart
import 'package:intl/intl.dart';

enum DatePrecision { exact, month, quarter }

String formatDateWithPrecision(DateTime date, DatePrecision precision) {
  switch (precision) {
    case DatePrecision.exact:
      return DateFormat('d MMMM yyyy', 'fr').format(date);
    case DatePrecision.month:
      return DateFormat('MMMM yyyy', 'fr').format(date);
    case DatePrecision.quarter:
      final q = ((date.month - 1) ~/ 3) + 1;
      return 'T$q ${date.year}';
  }
}

DatePrecision datePrecisionFromString(String? s) {
  switch (s) {
    case 'month': return DatePrecision.month;
    case 'quarter': return DatePrecision.quarter;
    default: return DatePrecision.exact;
  }
}

String datePrecisionToString(DatePrecision p) {
  switch (p) {
    case DatePrecision.exact: return 'exact';
    case DatePrecision.month: return 'month';
    case DatePrecision.quarter: return 'quarter';
  }
}
